src/pages/LoginPage/index.tsx

```diff
@@ -0,0 +1,29 @@
+import React from "react";
+import KeycloakContext from "../../contexts/user";
+import { Redirect } from "react-router-dom";
+
+function LoginPage() {
+    const [state, setState] = React.useContext(KeycloakContext);
+
+    if (!state.keycloak || state.keycloak.authenticated !== true) {
+        const keycloak = state.keycloak;
+        keycloak.init({onLoad: 'login-required'})
+            .success((authenticated: boolean) => {
+                console.log('Authenticated: ' + authenticated);
+                setState(({
```

Comment:
[The error is allegedly due to this anonymous function](https://imgur.com/8VjM2kT)

Reply:
```
index.js:1375 The above error occurred in the <Context.Provider> component:
    in Route (at PrivateRoute.tsx:13)
    in PrivateRoute (at Router.tsx:10)
    in div (at Router.tsx:9)
    in Router (at App.tsx:14)
    in KeycloakProvider (at App.tsx:13)
    in Router (created by BrowserRouter)
    in BrowserRouter (at App.tsx:12)
    in App (at src/index.tsx:7)
```

Reply:
```
import React from "react";
import {Route, RouteProps} from 'react-router-dom'
import UserContext from "../contexts/user";
import LoginPage from "../pages/LoginPage";

function PrivateRoute(props: RouteProps, component: React.ComponentType) {
    const [state] = React.useContext(UserContext);

    const authorizedDestination: React.ComponentType = (!state.keycloak || state.keycloak.authenticated !== true)
        ? LoginPage
        : component;

    return <Route {...props} component={ authorizedDestination } />
}

export default PrivateRoute;
```